include/boost/math/tools/differential_evolution.hpp

```diff
@@ -0,0 +1,261 @@
+        oss << " For infinite bounds, emulate with std::numeric_limits<Real>::lower() or use a standard infinite->finite transform.";
+        throw std::domain_error(oss.str());
+      }
+      if (!isfinite(bound[1])) {
+        oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+        oss << ": The upper bound must be finite, but got " << bound[1] << "."; 
+        oss << " For infinite bounds, emulate with std::numeric_limits<Real>::max() or use a standard infinite->finite transform.";
+        throw std::domain_error(oss.str());
+      }
+    }
+    if (NP_ < 4) {
+      oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+      oss << ": The population size must be at least 4.";
+      throw std::invalid_argument(oss.str());
+    }
+    if (threads_ > NP_) {
+        oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+        oss << ": There must be more individuals in the population than threads.";
+        throw std::invalid_argument(oss.str());
+    }
+    // From: "Differential Evolution: A Practical Approach to Global Optimization (Natural Computing Series)"
+    // > The scale factor, F in (0,1+), is a positive real number that controls the rate at which the population evolves.
+    // > While there is no upper limit on F, effective values are seldom greater than 1.0.
+    // ...
+    // Also see "Limits on F", Section 2.5.1:
+    // > This discontinuity at F = 1 reduces the number of mutants by half and can result in erratic convergence...
+    if (isnan(F_) || F_ >= Real(1) || F_ <= Real(0)) {
+      oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+      oss << ": F in (0, 1] is required, but got F=" << F << ".";
+      throw std::domain_error(oss.str());
+    }
+    if (max_generations_ < 1) {
+       oss << __FILE__ << ":" << __LINE__ << ":" << __func__;
+       oss << ": There must be at least one generation.";
+       throw std::invalid_argument(oss.str());
+    }
+  }
+
+  template <typename ArgumentContainer, class Func, class URBG>
+  ArgumentContainer
+  argmin(const Func cost_function, URBG &g,
+           ArgumentContainer* initial_guess= nullptr,
+           std::invoke_result_t<Func, ArgumentContainer> target_value = std::numeric_limits<std::invoke_result_t<Func, ArgumentContainer>>::quiet_NaN(),
+           std::atomic<bool> *cancellation = nullptr,
+           std::vector<std::pair<ArgumentContainer, std::invoke_result_t<Func, ArgumentContainer>>>
+               *queries = nullptr,
+          std::atomic<std::invoke_result_t<Func, ArgumentContainer>>* current_minimum_cost = nullptr
+         ) {
+    constexpr bool has_resize = requires(ArgumentContainer& t) {
+        t.resize(std::declval<typename ArgumentContainer::size_type>());
+    };
+    using ResultType = std::invoke_result_t<Func, ArgumentContainer>;
+    using std::clamp;
+    using std::round;
+    using std::isnan;
+    using Size_t = ArgumentContainer::size_type;
```

Comment:
I would rename this to avoid the confusion of just differing by one capital letter. Maybe `container_size_t`?